src/pip/_internal/vcs/bazaar.py

```diff
@@ -6,8 +6,15 @@
 from pip._vendor.six.moves.urllib import parse as urllib_parse
 
 from pip._internal.utils.misc import display_path, path_to_url, rmtree
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 from pip._internal.vcs.versioncontrol import VersionControl, vcs
 
+if MYPY_CHECK_RUNNING:
+    from typing import Optional, Tuple
+    from pip._internal.vcs.versioncontrol import RevOptions
+
+    AuthInfo = Tuple[Optional[str], Optional[str]]
```

Comment:
Instead of repeating this definition in each module, you can import it from `versioncontrol.py`.